src/other/tables.jl

```diff
@@ -0,0 +1,25 @@
+using Tables
+
+Tables.schema(df::DataFrame) = NamedTuple{Tuple(names(df)), Tuple{eltypes(df)...}}
+Tables.AccessStyle(df::DataFrame) = Tables.ColumnAccess()
+Tables.rows(df::DataFrame) = eachrow(df)
+Tables.columns(df::DataFrame) = df
+
+function _isiterabletable(x::T) where {T}
+    Base.isiterable(x) || return false
+    return Base.IteratorEltype(x) === Base.HasEltype() ? eltype(x) <: NamedTuple : false
+end
+
+function DataFrame(x::T) where {T}
+    if Tables.istable(T)
+        DataFrame([collect(u) for u in Tables.columns(x)], collect(Tables.names(Tables.schema(x))))
+    elseif _isiterabletable(x)
+        _DataFrame(x)
```

Comment:
@davidanthoff, I still need to do thorough testing, but I think this might do the trick for IterableTables. Essentially, we first do a check for `Tables.istable` (table types have to explicitly opt-in after https://github.com/JuliaData/Tables.jl/pull/9), then we'll do the `isiterabletable` check and if so, call `_DataFrame` that is defined in IterableTables. That should make it safe to remove the catchall definition from IterableTables.

Reply:
Ok, had to refactor the initial attempt a bit. But it seems to be working now. Companion PR to IterableTables is here: https://github.com/queryverse/IterableTables.jl/pull/84

Reply:
I think the ``Tables.istable`` helps a lot!

The ``_isiterabletable`` though won't work. It might seem to work now, but it will break with the next Query.jl release. I have two suggestions: first, I think it would be more efficient to call ``isiterabletable`` from TableTraits.jl directly here. The ``getiterator`` call can be costly and do a lot of stuff, and the ``isiterabletable`` call cann be made directly on ``T``, not the return value of ``getiterator``. It would also force us to come up with a solution for the case where ``isiterabletable`` returns ``missing``, which is the case where we don't know whether it is an iterable table until we start iterating it. That will complicate things, but I don't see any way around it, unfortunately... I think maybe the general strategy would be: if ``isiterabletable`` returns ``true``, everything is easy. If it returns ``false``, also easy, we just go to the ``convert`` call right away. If it returns ``missing``, we next can check whether ``T`` is an iterable with ``IteratorInterfaceExtensions.isiterable``. If that returns ``false``, we go to the ``convert`` case. If that returns ``true``, we will have to start iterating, look at the first element and then decide whether this is an iterable table or not... Ugly, ugly, but I can't think of anything to avoid that.